pkg/requestor.go

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ * Copyright 2021 Intel Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Tingyu Zeng, Dell
+ *******************************************************************************/
+
+package pkg
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+
+	"github.com/edgexfoundry/go-mod-secrets/v2/pkg/token/fileioperformer"
+)
+
+type HTTPSRequester interface {
+	Insecure() Caller
+	WithTLS(io.Reader, string) Caller
+}
+
+type fluentRequester struct {
+	logger logger.LoggingClient
+}
+
+func NewRequester(logger logger.LoggingClient) HTTPSRequester {
+	return &fluentRequester{logger}
+}
+
+func (r *fluentRequester) Insecure() Caller {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	return &http.Client{Timeout: 10 * time.Second, Transport: tr}
```

Comment:
prefer to define a constant for this `Timeout`:
```suggestion
	return &http.Client{Timeout: httpClientTimeoutDuration, Transport: tr}
```
where httpClientTimeoutDuration = 10 * time.Second